pkg/operator/configobservation/node/observe_latency_profile_test.go

```diff
@@ -0,0 +1,102 @@
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+
+	"github.com/openshift/cluster-kube-controller-manager-operator/pkg/operator/configobservation"
+)
+
+type workerLatencyProfileTestScenario struct {
+	name                   string
+	existingConfig         map[string]interface{}
+	expectedObservedConfig map[string]interface{}
+	workerLatencyProfile   configv1.WorkerLatencyProfileType
+}
+
+func multiScenarioLatencyProfilesTest(
+	configObserveFn func(configobserver.Listers, events.Recorder, map[string]interface{}) (ret map[string]interface{}, errs []error),
+	scenarios []workerLatencyProfileTestScenario,
+) func(*testing.T) {
+	return func(t *testing.T) {
+		for _, scenario := range scenarios {
+			t.Run(scenario.name, func(t *testing.T) {
+				// test data
+				eventRecorder := events.NewInMemoryRecorder("")
+				configNodeIndexer := cache.NewIndexer(cache.MetaNamespaceKeyFunc, cache.Indexers{})
+				configNodeIndexer.Add(&configv1.Node{
+					ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
+					Spec:       configv1.NodeSpec{WorkerLatencyProfile: scenario.workerLatencyProfile},
+				})
+				listers := configobservation.Listers{
+					NodeLister: configlistersv1.NewNodeLister(configNodeIndexer),
+				}
+
+				// act
+				observedKubeAPIConfig, err := configObserveFn(listers, eventRecorder, scenario.existingConfig)
+
+				// validate
+				if len(err) > 0 {
+					t.Fatal(err)
+				}
+				if !cmp.Equal(scenario.expectedObservedConfig, observedKubeAPIConfig) {
+					t.Fatalf("unexpected configuration, diff = %v", cmp.Diff(scenario.expectedObservedConfig, observedKubeAPIConfig))
+				}
+			})
+		}
+	}
+}
+
+func TestObserveNodeMonitorGracePeriod(t *testing.T) {
+	scenarios := []workerLatencyProfileTestScenario{
+		// scenario 1: empty worker latency profile
+		{
+			name:                   "default value is not applied when worker latency profile is unset",
+			expectedObservedConfig: nil,
+			workerLatencyProfile:   "", // empty worker latency profile
+		},
+
+		// scenario 2: Default
+		{
+			name: "worker latency profile Default: config with node-monitor-grace-period=40s",
+			expectedObservedConfig: map[string]interface{}{
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"40s"},
+				},
+			},
+			workerLatencyProfile: configv1.DefaultUpdateDefaultReaction,
+		},
+
+		// scenario 3: MediumUpdateAverageReaction
+		{
+			name: "worker latency profile MediumUpdateAverageReaction: config with node-monitor-grace-period=2m",
+			expectedObservedConfig: map[string]interface{}{
+				"extendedArguments": map[string]interface{}{
+					"node-monitor-grace-period": []interface{}{"2m0s"},
+				},
+			},
+			workerLatencyProfile: configv1.MediumUpdateAverageReaction,
+		},
+
+		// scenario 4: LowUpdateSlowReaction
```

Comment:
I suppose you can add also scenario for unknown profile

Reply:
Added the scenario for unknown profile as well.